### PR TITLE
Feature/alignment refactor interface

### PR DESCRIPTION
**Pull request by hroest**
switch interface from ConsensusMap to std::vector<Peak2D> as this is a much cleaner interface and the algorithm only needs RT, m/z and int information. This potentially allows other code to use an easy interface for alignment that takes two simple 3D peaklist as input.

### CODE REVIEW

File: src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringAffineSuperimposer.cpp [hunk 1]
```diff
@@ -840,43 +812,49 @@ namespace OpenMS
...
+      // sort the last data points by ascending intensity (from the right, using reverse iterators)
+      //  -> linear in complexity, should be faster than sorting and then taking cutoff
+      //  TODO code-review this
+      if (model_map.size() > num_used_points)
       {
-        model_map_ini.sortByIntensity(true);
-        model_map_ini.resize(num_used_points);
+        std::nth_element(model_map.rbegin(), model_map.rbegin() + (model_map.size() - num_used_points),
+            model_map.rend(), Peak2D::IntensityLess());
+        model_map.resize(num_used_points);
```
Comment (hroest) [inline src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringAffineSuperimposer.cpp]: can somebody else comment on this, I believe it does what it is supposed to do: only keep the n most intense peaks

Comment (hroest) [inline src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringAffineSuperimposer.cpp]: basically, `Peak2D::IntensityLess` sorts by increasing intensity, so we start sorting the vector from the back. `std::nth_element` guarantees that all elements before this point are less than or equal to all elements after

Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringAffineSuperimposer.cpp]: I am not sure: "The other elements are left without any specific order, except that none of the elements preceding nth are greater than it, and none of the elements following it are less."
so I think there is no strict sorting implied.
Maybe http://www.cplusplus.com/reference/algorithm/partial_sort/ ?

Comment (hroest) [inline src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringAffineSuperimposer.cpp]: yes, but we dont *need* sorting. we just want to keep the _n_ most intense elements  and remove the rest (in this case n=num_used_points). As long as we are guaranteed that all elements we want to keep are one one side and all the ones we want to throw away are on the other side of our cutoff, we are fine. Afterwards the array is sorted again by m/z

Comment (timosachsenberg) [inline src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringAffineSuperimposer.cpp]: OK I see. Seems correct to me then.

Comment (hroest) [inline src/openms/source/ANALYSIS/MAPMATCHING/PoseClusteringAffineSuperimposer.cpp]: to be sure, I will add a test :-)

**Comment by hroest** [comment]
ok, added a test for using different number of pts. seems to work!

**Comment by timosachsenberg** [comment]
+1

**Pull request merged**